content/blog/2019/10/2019-10-04-vscode-caseStudy.adoc

```diff
@@ -0,0 +1,71 @@
+tags:
+- community-bridge
+- JCasC
+- VSCode
+- community
+author: sladyn98
+---
+
+
+==== Configuration-as-code plugin
+The link:https://plugins.jenkins.io/configuration-as-code[Configuration as Code plugin] has been designed as an opinionated way to configure Jenkins based on human-readable declarative configuration files. Writing such a file should be feasible without being a Jenkins expert, just translating into code a configuration process one is used to executing in the web UI. The plugin uses a schema to verify the files being applied to the Jenkins instance.
+
+With the new JSON Schema being enabled developers can now test their yaml file against it. The new schema checks for the configurators, their types and to a certain extent the nesting of the configurators. VSCode allows us to test out the schema right out of the box with some modifications.
+
+=== Steps to Enable the Schema Validation
+
+a) The first step includes installing the YAML plugin from Red Hat. Open up the Extension list in VSCode editor using `Ctrl + Shift + X`.
+
+b) Select and install the link:https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml[YAML plugin] by Red Hat. Reload the editor if necessary.
+
+c) The next step includes inserting the JSON Schema inside the VSCode workspace. We would be including the schema in the current directory.
+
+d) Spin up your Jenkins server and navigate to `/configuration-as-code/jsonSchema`. 
+
+e) Copy and paste this schema into a new file called `json.schema`.
+
+f) In order to enable validation we need to include it in the workspace settings. In order to do that
+navigate to `File` and then `Preference` and then `Settings`. Inside settings search for `json` and inside `settings.json` include the following configuration.
+
+[source, json]
+----
+{
+"yaml.schemas": {
+        "schema.json": "y[a]?ml"
+    }
+}
+----
+
+You can specify a glob pattern as the value for schema.json which is the file name for the schema. This would apply the schema to all yaml files. eg: `.[y[a]?ml]`
+
+g) The following tasks can be done using VSCode:
+
+  a) Auto completion (Ctrl + Space):
+    Auto completes on all commands.
+
+  b) Document Outlining (Ctrl + Shift + O):
+  Provides the document outlining of all completed nodes in the file.
+
+h) Create a new file under the work directory called `jenkins.yml`. For example consider the following contents for the file:
+
+[source, yaml]
+----
+jenkins:
+  systemMessage: “Hello World”
+  numExecutors:2
+----
+
+i) The above yaml file is valid according to the schema and vscode should provide you with validation and autocompletion for the same.
+
+=== Screenshots
+
+image:/images/projects/jcasc/vscode.png[]
+
+
+For any suggestions or dicussions regarding the schema feel free to join our gitter link:https://gitter.im/jenkinsci/jcasc-devtools-project[channel].
+Issues can be created on link:https://github.com/jenkinsci/configuration-as-code-plugin/issues[here].
```

Comment:
using the text 'here' in a link is an accessibility anti-pattern, it gives no context to the reader on where they will be going without looking at the link
```suggestion
Issues can be created on link:https://github.com/jenkinsci/configuration-as-code-plugin/issues[GitHub].
```

Reply:
:+1: